tests/test_sshfsmount.cpp

```diff
@@ -241,3 +247,107 @@ TEST_F(SshfsMount, unblocks_when_sftpserver_exits)
     mount.join();
     EXPECT_TRUE(stopped_ok);
 }
+
+TEST_F(SshfsMount, throws_when_unable_to_change_dir)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"cd"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    EXPECT_THROW(make_sshfsmount(), std::runtime_error);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, throws_when_unable_to_get_current_dir)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"pwd"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    EXPECT_THROW(make_sshfsmount(), std::runtime_error);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, executes_commands)
+{
+    std::unordered_map<std::string, std::string> commands = {
```

Comment:
The order of those actually matters, think you could easily make it so that this fails if commands are run out of order?

I suppose it still makes sense to ignore calls that we don't include in the map, but if we're interested about a call, ideally we'd check the order.

Reply:
Done.

Reply:
Sure, done